File: WorldCupApp/app/src/main/java/com/example/worldcupapp/MainActivity.java
package com.example.worldcupapp;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import com.example.worldcupapp.databinding.ActivityMainBinding;

import java.util.ArrayList;
import java.util.List;
import java.util.function.Function;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        setupListView();
    }
    private void setupListView()
    {
        CustomArrAdapter adapter = new CustomArrAdapter(this,getData());
        binding.listView.setAdapter(adapter);
        adapter.clickListener =this::setAdapterOnClickListener;
    }

    private List<CountryModel> getData()
    {
        ArrayList<CountryModel> countries = new ArrayList<CountryModel>();
        CountryModel brazil = new CountryModel("Brazil", 5, R.drawable.brazil );
        CountryModel france = new CountryModel("France", 1, R.drawable.france);
        CountryModel germany = new CountryModel("Germany", 6, R.drawable.germany);
        CountryModel saudi = new CountryModel("Saudi", 2, R.drawable.saudiarabia);
        CountryModel usa = new CountryModel("United States America", 10, R.drawable.unitedstates);
        CountryModel uk = new CountryModel("United Kingdom", 4, R.drawable.unitedkingdom);

        countries.add(brazil);
        countries.add(germany);
        countries.add(france);
        countries.add(saudi);
        countries.add(usa);
        countries.add(uk);

        return countries;

    }

    private View.OnClickListener setAdapterOnClickListener(CountryModel data)
    {
        View.OnClickListener listener = v -> Toast.makeText(v.getContext(), data.toString(), Toast.LENGTH_LONG).show();

        return  listener;
    }
}

File: CleanArchMovie/app/src/main/java/com/example/simplemovieapp/presentation/home/MainActivity.kt
package com.example.simplemovieapp.presentation.home

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import com.example.simplemovieapp.R
import com.example.simplemovieapp.data.models.movie.Movie
import com.example.simplemovieapp.databinding.ActivityMainBinding
import com.example.simplemovieapp.presentation.artist.ArtistsActivity
import com.example.simplemovieapp.presentation.movie.MovieActivity
import com.example.simplemovieapp.presentation.tv.TvActivity

class MainActivity : AppCompatActivity() {
    private lateinit var binding : ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        setupOnClickListener()
    }

    private fun setupOnClickListener() {
        binding.btnMovie.setOnClickListener {
            var intent = Intent(this, MovieActivity::class.java);
            startActivity(intent)
        }

        binding.btnTv.setOnClickListener {
            var intent = Intent(this, TvActivity::class.java);
        }

        binding.btnArtists.setOnClickListener {
            var intent = Intent(this, ArtistsActivity::class.java);
        }
    }
}